src/monster.cpp

```diff
@@ -2166,12 +2166,22 @@ void monster::die( Creature *nkiller )
             ch->rem_morale( MORALE_KILLER_NEED_TO_KILL );
         }
     }
-    // We were tied up at the moment of death, add a short rope to inventory
-    if( has_effect( effect_tied ) ) {
-        if( tied_item ) {
-            add_item( *tied_item );
-            tied_item = cata::nullopt;
-        }
+    // Drop items stored in optionals
+    if( tack_item ) {
```

Comment:
I suggest putting those 4 lines into a separate function, so you only need to call `the_function( tack_item );` and so on.

Reply:
Given that this is the only place it would be used I think it would be worse for readability to move these to a function.

Reply:
This contains the same code repeated 4 times. Converting it to 4 simple function calls is much shorter and much easier to read.